src/hotspot/share/cds/aotMetaspace.cpp

```diff
@@ -1056,7 +1057,21 @@ void AOTMetaspace::exercise_runtime_cds_code(TRAPS) {
   CDSProtectionDomain::to_file_URL("dummy.jar", Handle(), CHECK);
 }
 
+bool AOTMetaspace::preimage_static_archive_dumped() {
+  assert(CDSConfig::is_dumping_preimage_static_archive(), "Required");
+  return _preimage_static_archive_dumped == 1;
```

Comment:
Should it be  AtomicAccess::load(&_preimage_static_archive_dumped) here?

Reply:
Andrew brought this up also; I'll add but should it be AtomicAccess::load or AtomicAccess::load_acquire?